include/filters/filter_base.hpp

```diff
@@ -0,0 +1,279 @@
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef FILTERS_FILTER_BASE_H_
+#define FILTERS_FILTER_BASE_H_
+
+#include <typeinfo>
+#include "rclcpp/rclcpp.hpp"
+
+namespace filters
+{
+
+/** \brief A Base filter class to provide a standard interface for all filters
+ *
+ */
+template<typename T>
+class FilterBase
+{
+public:
+  /** \brief Default constructor used by Filter Factories
+   */
+  FilterBase():configured_(false){};
+
+  /** \brief Virtual Destructor
+   */
+  virtual ~FilterBase(){};
+
+  /** \brief Configure the filter from the parameter server 
+   * \param The parameter from which to read the configuration
+   * \param node_handle The optional node handle, useful if operating in a different namespace.
+   */
+  bool configure(
+    const std::string& param_prefix,
+    const std::string& filter_name,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr & node_logger,
+    const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr & node_params)
+  {
+    if (configured_)
+    {
+      RCLCPP_WARN(
+        node_logger->get_logger(),
+        "Filter %s already being reconfigured",
+        filter_name_.c_str());
+    };
+    if (!node_params) {
+      throw std::runtime_error("Need a parameters interface to function.");
+    }
+
+    configured_ = false;
+    bool retval = true;
+
+    filter_name_ = filter_name;
+    param_prefix_ = param_prefix;
+    params_interface_ = node_params;
+    logging_interface_ = node_logger;
+
+    // Make param_prefix an empty string or a string ending in '.'
+    if (!param_prefix_.empty()) {
+      if ('.' != param_prefix_.back()) {
+        param_prefix_ += '.';
+      }
+    }
+
+    retval = retval && configure();
```

Comment:
`retval` can't be changed between where it is defined above and here, so I'd suggest:

```suggestion
    bool retval = configure();
```

Reply:
got rid of retval in 8dba039